app/src/main/java/com/palone/planahead/screens/home/ui/components/TaskItem.kt

```diff
@@ -29,108 +38,178 @@ import com.palone.planahead.data.database.alert.properties.AlertType
         Row(
-            modifier = modifier.clickable { shouldExpand.value = !shouldExpand.value },
-            horizontalArrangement = Arrangement.SpaceBetween,
+            modifier = Modifier
+//                .clickable { shouldExpand.value = !shouldExpand.value }
+                .fillMaxWidth(),
+            horizontalArrangement = Arrangement.Start,
             verticalAlignment = Alignment.CenterVertically
         ) {
-            Row(verticalAlignment = Alignment.CenterVertically) {
-                Icon(
-                    Icons.Default.BrokenImage,
-                    "Image of selected icon",
-                    modifier = Modifier.size(50.dp)
+            TaskTriggerPillar(hasPassed)
+            Column(modifier = Modifier.padding(10.dp)) {
+                Text(
+                    text = task.description,
+                    fontWeight = FontWeight.Bold,
+                    fontSize = 20.sp
                 )
-                Column {
-                    Text(text = task.description)
-                    alertTypes.forEach { alertType ->
-                        Text(text = alertType.name, fontSize = 5.sp)
-                    }
-                }
-            }
-            Text(text = task.taskType.name)
-            Column {
-                alerts.forEach {
-                    val date = it.eventMillisInEpoch?.let { it1 -> Date(it1) }
-                    val pattern = "yyyy-MM-dd HH:mm:ss"
-                    val simpleDateFormat = SimpleDateFormat(pattern, Locale.getDefault())
-                    val formattedDate = date?.let { it1 -> simpleDateFormat.format(it1) }
-                    if (formattedDate != null) {
-                        Text(text = formattedDate)
+                EventDate(
+                    task,
+                    isToday,
+                    taskDateTime,
+                    hasPassed,
+                    isTomorrow
+                )
+                Row {
+                    alerts.forEach {
+                        AlertItem(alert = it, task = task)
                     }
+                    Text(text = " | Priority:  ${task.priority}")
                 }
             }
-            if (shouldExpand.value) {
-                Icon(Icons.Default.ArrowDropDown, "Trailing Icon")
-            } else {
-                Icon(Icons.Default.ArrowRight, "Trailing Icon")
-            }
-        }
-        if (shouldExpand.value) {
-            ExpandedItem(task = task, alerts = alerts, onDelete = onDelete)
         }
     }
 }
 
 @Composable
-fun ExpandedItem(
+private fun TaskTriggerPillar(hasPassed: Boolean) {
+    Card(
+        modifier = Modifier
+            .width(25.dp)
+            .fillMaxHeight(),
+        shape = RoundedCornerShape(0.dp),
+        colors = CardDefaults.cardColors(containerColor = if (hasPassed) Color.Red else Color.Cyan)
+    ) {
+        Icon(Icons.Default.Timer, null)
+    }
+}
+
+@Composable
```

Comment:
There are way too many components here. We should split them into separate files.